utils/s3/client.cc

```diff
@@ -204,7 +204,7 @@ future<semaphore_units<>> client::claim_memory(size_t size) {
 }
 
 client::group_client::group_client(std::unique_ptr<http::experimental::connection_factory> f, unsigned max_conn, const aws::retry_strategy& retry_strategy)
-    : retryable_client(std::move(f), max_conn, map_s3_client_exception, http::experimental::client::retry_requests::yes, retry_strategy) {
+    : retryable_client(std::move(f), max_conn, map_s3_client_exception, http::experimental::client::retry_requests::no, retry_strategy) {
```

Comment:
This line disables http client retries for _any_ usage, not just for chunked_download_source. From the commit message it seems that it only affects the latter.

Also, if disabling http retries, need to explain why it's safe. Presumably S3 client already handles transport errors its own? How?